### PR TITLE
(PCP-384) Add `/pcp-broker/connect` authz check for association

### PR DESCRIPTION
**Pull request by MikaelSmith**
Adds a `/pcp-broker/connect` authorization check via tk-auth for
association requests.

Also changes tests to use shorter timeouts when expecting no response.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
@mruzicka updated with `associate_request` and `inventory_request` mapping to `/pcp-broker/send` and using authorization.

File: src/puppetlabs/pcp/broker/core.clj [hunk 1]
```diff
@@ -287,60 +291,65 @@
...
+      (let [uris (doall (filter (partial get-websocket broker) ((:find-clients broker) (:query data))))
+            response-data {:uris uris}
+            response (-> (message/make-message :message_type "http://puppetlabs.com/inventory_response"
+                                               :targets [(:sender message)]
+                                               :in-reply-to (:id message)
+                                               :sender "pcp:///server")
+                         (message/set-expiry 3 :seconds)
+                         (message/set-json-data response-data))]
+        (s/validate p/InventoryResponse response-data)
+        (accept-message-for-delivery broker (capsule/wrap response)))))
```
Comment (parisiale) [inline src/puppetlabs/pcp/broker/core.clj]: You'll end up calling `authorized?` again.

**Comment by parisiale** [comment]
I'm not sure we should be invoking `authorized?` in `accept-message-for-delivery`.

File: doc/authorization.md [hunk 2]
```diff
@@ -69,3 +69,110 @@ authorization: {
...
+
+  ]
+}
+```
+
+### Inventory Request
+
+Not all nodes need or should have access to the full inventory of nodes connected to a PCP broker.
+Inventory requests are one way of acquiring that information; another functionally equivalent way
+to discover all connected nodes is a message using a wildcard to specify the target.
```
Comment (parisiale) [inline doc/authorization.md]: Do you mean, send a multicast message and use the endpoint responses to determine the inventory?
Why don't you mention the `destination_report` flag?

File: doc/authorization.md [hunk 3]
```diff
@@ -69,3 +69,110 @@ authorization: {
...
+
+Both types of requests can be matched by trapperkeeper-authorization with the following
```
Comment (parisiale) [inline doc/authorization.md]: You have inventory requests. You have PXP requests. But, in general, PCP messages are not requests.

**Comment by parisiale** [comment]
More in general, can we only authorize messages at reception?

File: src/puppetlabs/pcp/broker/core.clj (hunk 1 above)
Comment (MikaelSmith) [inline src/puppetlabs/pcp/broker/core.clj]: It calls `authorized?` on the response, where-as above calls it on the request. `authorized?` and `accept-message-for-delivery` get a bit entangled.

**Comment by MikaelSmith** [comment]
`accept-message-for-delivery` authorizes messages because they're being forwarded to other nodes. We could move that. I'm not sure about whether we should authorize other delivery, I'll think about it.

**Comment by parisiale** [comment]
👍

**Pull request merged**